.github/workflows/lint.yml

```diff
@@ -52,5 +52,9 @@ jobs:
     timeout-minutes: 5
     steps:
       - uses: actions/checkout@v4
+      - name: set up go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ inputs.go-version }}
```

Comment:
since we are editing these files anyways, might as well also make the improvement to use the go.mod file.

https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Then we can eliminate the `inputs.go-version` as well. This applies to all the use of go versions in the CI.